app/scripts/modules/core/src/pipeline/config/stages/managed/ImportDeliveryConfigExecutionDetails.tsx

```diff
@@ -2,10 +2,69 @@ import React from 'react';
 
 import { ExecutionDetailsSection, IExecutionDetailsSectionProps, StageFailureMessage } from 'core/pipeline';
 import { IGitTrigger } from 'core/domain';
+import { CollapsibleSection, Markdown } from 'core/presentation';
 import { SETTINGS } from 'core/config';
 
 const NOT_FOUND = 'Not found';
 
+// Turn off linter for the block below because it wants CamelCase property names.
+/* eslint-disable @typescript-eslint/camelcase */
+const ERROR_MESSAGE_MAP: any = {
+  missing_property: 'The following property is missing:',
+  invalid_type: 'The type of the following property is invalid:',
+  invalid_format: 'The format of the following property is invalid:',
+  invalid_value: 'The value of the following property is invalid:',
+};
+/* eslint-enable */
+
+interface IDeliveryConfigImportErrorDetails {
+  error: string;
+  message: string;
+  pathExpression: string;
+}
+
+interface IDeliveryConfigImportError {
+  message: string;
+  details?: IDeliveryConfigImportErrorDetails;
+}
+
+const CustomErrorMessage = (message: string, debugDetails?: string) => {
+  return (
+    <div>
+      <div className="alert alert-danger">
+        <b>There was an error parsing your delivery config file.</b>
+        <br />
+        <Markdown message={message} style={{ wordBreak: 'break-all' }} />
+        <br />
+        {debugDetails && (
+          <CollapsibleSection heading={({ chevron }) => <span>{chevron} Debug Details</span>}>
+            <pre style={{ whiteSpace: 'pre-wrap' }}>{debugDetails}</pre>
+          </CollapsibleSection>
+        )}
+      </div>
+    </div>
+  );
+};
+
+function buildCustomErrorMessage(error: IDeliveryConfigImportError) {
```

Comment:
Let's stay consistent with function syntax within the same file:

```suggestion
const buildCustomErrorMessage = (error: IDeliveryConfigImportError) => {
```

Reply:
I wrote this in both formats but weren't sure one was more idiomatic than the other. What's the advantage of the `const` form?

Reply:
there's not really a huge difference in practice most of the time, we just tend to stick to one style when we're in the same file/module.

the biggest different between the two listed here is that the `const` version is using an arrow function (`() => {}`), which binds the function to the "lexical" `this` rather than giving each function invocation its own fresh `this` binding. If that doesn't make any sense to you right now, it's because old school JS's `this` semantics were super confusing and basically made no sense 😆. Even though I'm using an arrow function in this example, you could also assign a regular `function` to a `const` binding like `const buildErrorMessage = function buildErrorMessage() {};`. That has it's own bizarre differences ('hoisting' behavior is different), but I recommend you just worry about that right now! 😉